File: RR-Frontend/src/app/core/models/shared/finding/finding.base.ts
import { IDeletable } from '@interfaces/deletable.interface';
import { ModelBase } from '@models/model-base';
import { Guid } from '../guid/guid';

export class FindingBase extends ModelBase implements IDeletable {
  isDeleted: boolean;
  number: number;
  validationId?: string;
  scopeId: string;
  constructor() {
    super();
    this.isDeleted = false;
    this.number = 1;
    this.validationId = Guid.create();
    this.scopeId = Guid.create();
  }
}

File: RR-Backend/RadioReport.Common/RadioReport.Common.Logic/Models/FindingBase.cs
using RadioReport.Common.Logic.Interfaces;
using System;

namespace RadioReport.Common.Logic.Models
{
    /// <summary>
    /// Base finding model entity class. From it should derive all finding models in RR project
    /// </summary>
    public abstract class FindingBase : ModelBase, ICollectionItem
    {
        public virtual byte Number { get; set; }

        public abstract string InstanceName { get; }

        public virtual bool IsDeleted { get; set; }

        public virtual byte? ParentNumber { get; set; }

        public Guid? ScopeId
        {
            get => scopeId.HasValue ? scopeId : Id;
            set => scopeId = value;
        }

        private Guid? scopeId;
    }
}
